etcdserver/server_test.go

```diff
@@ -823,6 +826,94 @@ func TestTriggerSnap(t *testing.T) {
 	}
 }
 
+// TestConcurrentApplyAndSnapshot will send out snapshots concurrently with
+// proposals.
+func TestConcurrentApplyAndSnapshot(t *testing.T) {
```

Comment:
We have different path for v2 and v3 snapshot code path. Probably we should mention this only applies for v3?